packages/cyberstorm/src/components/Breadcrumb.tsx

```diff
@@ -0,0 +1,86 @@
+import { Fragment, FunctionComponent, ReactNode } from "react";
+
+import { faHouse, faSlashForward } from "@fortawesome/pro-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import styles from "./componentStyles/Breadcrumb.module.css";
+import { IndexLink } from "./Links";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+
+type Label = ReactNode | string;
+
+type Crumb<P> =
+  | {
+      label: Label;
+      LinkComponent?: FunctionComponent;
+    }
+  | {
+      label: Label;
+      LinkComponent: FunctionComponent<P>;
+      LinkProps: P;
+    };
+
+interface BreadCrumbProps<P> {
+  parts: Crumb<P>[];
+}
+
+export function BreadCrumb<P>(props: BreadCrumbProps<P>) {
+  if (!props.parts) {
+    return null;
+  }
+  const parts = [...props.parts];
+  const leaf = parts.pop();
+
+  return (
+    <div className={styles.root}>
+      {parts.map((crumb, i) => (
+        <Fragment key={`${crumb.label}-${i}`}>
+          <MidCrumb crumb={crumb} />
+          <Separator />
+        </Fragment>
+      ))}
+
+      <LeafCrumb crumb={leaf} />
+    </div>
+  );
+}
+
+function MidCrumb<P>(props: { crumb: Crumb<P> }) {
+  const { crumb } = props;
+  const label = CrumbLabel(crumb.label);
+  return (
+    <crumb.LinkComponent {...crumb.LinkProps}>{label}</crumb.LinkComponent>
```

Comment:
In discord, you had asked about two issues on this line.

The first one, regarding `crumb.LinkComponent`: L14, defines that `LinkComponent` can be `undefined`, but here it's not checked whether it actually is `undefined`, instead it's treated as a function component. To fix this perhaps add a if-check that just returns `label` if the component is `undefined`? (Although IDK if it makes sense for the `MidCrumb` to consist only of label? Might be some use cases for it 🤷🏻)

The second one, regarding `...crumb.LinkProps`: Currently `MidCrumb` doesn't know whether the `crumb` it received is the one defined on L12-15 or the one on L16-L20. I.e. the `crumb` might not have `LinkProps` property at all. How to solve this seems a bit more complex but can probable be done using a type guard function. But before I spend more time on that, I need to know what was the goal of typing the crumbs like they've typed here (vs. as they've been typed on the old implementation)?

Reply:
Goal was to have more precise types for the `parts`, so that we can get rid of the `// eslint-disable-line @typescript-eslint/no-explicit-any`.

In the latest commits I've reworked the types a bit, so that there are at least no errors and everything works as intended. Though I'm not sure if the code is that "clean".

Tyvm for explaining the errors, idk why it is so hard for me to extract the actual problem out of those.